app/src/main/java/com/example/paritosh/stormy/WeatherActivity.java

```diff
@@ -10,70 +12,62 @@
 import android.widget.Toast;
 
 import com.example.paritosh.stormy.databinding.ActivityMainBinding;
-import com.example.paritosh.stormy.model.CurrentWeather;
 import com.example.paritosh.stormy.model.CurrentWeatherDataBindingModel;
 
-public class MainActivity extends AppCompatActivity {
+public class WeatherActivity extends AppCompatActivity implements WeatherContract.WeatherView {
 
-    private static final String TAG = MainActivity.class.getSimpleName();
-    private WeatherDataProvider weatherDataProvider = new WeatherDataProvider();
+    private WeatherPresenterIMPL presenter = new WeatherPresenterIMPL(this);
```

Comment:
Why create presenter separately?

private WeatherContract.WeatherPresenter weatherPresenter = new WeatherPresenterImpl(this);